sonic_platform_base/sfp_base.py

```diff
@@ -304,3 +304,38 @@ def set_power_override(self, power_override, power_set):
             False if not
         """
         raise NotImplementedError
+
+    def read_eeprom(self, offset, num_bytes):
```

Comment:
maybe we can add a bulk api, where you can have a list of [offset, number_bytes], so that it can read a few fields at once. same for the write_eeprom api.

also, do we need read/write a bit? is that supported by i2c bus?

Reply:
>maybe we can add a bulk api, where you can have a list of [offset, number_bytes], so that it can read a few fields at once. same for the write_eeprom api.

We would also need to add these bulk functions to the underlying layer. Do you foresee a performance issue if we do not support bulk transfers?

> also, do we need read/write a bit? is that supported by i2c bus?

No, the unit of transfer for I2C is one byte.